### PR TITLE
DRY out simple controller redirects

### PR DESCRIPTION
**Pull request by camillevilla**
## Summary
This refactor addresses #564 and replaces 

```
respond_to do |format|
  format.html { redirect_to moods_path }
  format.json { head :no_content }
end
```

with the `redirect_to_path` method in the following controllers:
- categories
- groups
- medications
- meetings
- moments
- moods
- search
- strategies

## Notes
Hi @julianguyen! I tried to make sure I "broke" tests by (1) commenting out redirects, (2) replacing existing code with `redirect_to_path` (3) making sure tests passed again. 

But there were a couple of places where controller test coverage could be improved. In particular :
- `medications_controller_spec.rb`
- `meetings_controller_spec.rb`
- `moments_controller_spec`

These could be good tickets to add to Improving Test Coverage, something like #534.

### CODE REVIEW

**Comment by julianguyen** [comment]
Could you create an issue for improving the test coverage into those specs you identified?

File: app/controllers/medications_controller.rb [hunk 1]
```diff
@@ -73,18 +67,12 @@ def update
   # DELETE /medications/1.json
   def destroy
     @medication.destroy
-    respond_to do |format|
-      format.html { redirect_to medications_path }
-      format.json { head :no_content }
-    end
+    redirect_to_path(medications)
```
Comment (HashNotAdam) [inline app/controllers/medications_controller.rb]: This one should be `medications_path`

Can you write a test that proves this doesn't work and then fix it?

**Comment by HashNotAdam** [comment]
@julianguyen Oh.. boo... you merged before I could write above

File: app/controllers/medications_controller.rb (hunk 1 above)
Comment (julianguyen) [reply app/controllers/medications_controller.rb]: Oh shoot yeah. Can you address this in another PR @camillevilla?

Comment (camillevilla) [reply app/controllers/medications_controller.rb]: whoops! will do @HashNotAdam @julianguyen

**Pull request merged**